packages/common/src/state/simple.ts

```diff
@@ -0,0 +1,185 @@
+ * have a look at the `@ethereumjs/statemanager` package.
+ */
+export class SimpleStateManager implements EVMStateManagerInterface {
+  public accountStack: Map<PrefixedHexString, Account | undefined>[] = []
+  public codeStack: Map<PrefixedHexString, Uint8Array>[] = []
+  public storageStack: Map<string, Uint8Array>[] = []
+
+  originalStorageCache: {
+    get(address: Address, key: Uint8Array): Promise<Uint8Array>
+    clear(): void
+  }
+
+  constructor() {
+    this.add()
+    this.originalStorageCache = new OriginalStorageCache(this.getContractStorage.bind(this))
+  }
+
+  topA() {
+    return this.accountStack[this.accountStack.length - 1]
+  }
+  topC() {
+    return this.codeStack[this.codeStack.length - 1]
+  }
+  topS() {
+    return this.storageStack[this.storageStack.length - 1]
+  }
+
+  add() {
+    const newTopA = new Map(this.topA())
+    for (const [address, account] of newTopA) {
+      const accountCopy =
+        account !== undefined
+          ? Object.assign(Object.create(Object.getPrototypeOf(account)), account)
+          : undefined
+      newTopA.set(address, accountCopy)
+    }
+    this.accountStack.push(newTopA)
+    this.codeStack.push(new Map(this.topC()))
+    this.storageStack.push(new Map(this.topS()))
+  }
+
+  async getAccount(address: Address): Promise<Account | undefined> {
+    return this.topA().get(address.toString())
+  }
+
+  async putAccount(address: Address, account?: Account | undefined): Promise<void> {
+    this.topA().set(address.toString(), account)
+  }
+
+  async deleteAccount(address: Address): Promise<void> {
+    this.topA().set(address.toString(), undefined)
+  }
+
+  async modifyAccountFields(address: Address, accountFields: AccountFields): Promise<void> {
+    let account = await this.getAccount(address)
+    if (!account) {
+      account = new Account()
+    }
+    account.nonce = accountFields.nonce ?? account.nonce
+    account.balance = accountFields.balance ?? account.balance
+    account.storageRoot = accountFields.storageRoot ?? account.storageRoot
+    account.codeHash = accountFields.codeHash ?? account.codeHash
+    await this.putAccount(address, account)
+  }
+
+  async getContractCode(address: Address): Promise<Uint8Array> {
+    return this.topC().get(address.toString()) ?? new Uint8Array(0)
+  }
+
+  async putContractCode(address: Address, value: Uint8Array): Promise<void> {
+    this.topC().set(address.toString(), value)
+    if ((await this.getAccount(address)) === undefined) {
+      await this.putAccount(address, new Account())
+    }
+    await this.modifyAccountFields(address, { codeHash: keccak256(value) })
+  }
+
+  async getContractCodeSize(address: Address): Promise<number> {
+    const contractCode = await this.getContractCode(address)
+    return contractCode.length
+  }
+
+  async getContractStorage(address: Address, key: Uint8Array): Promise<Uint8Array> {
+    return this.topS().get(`${address.toString()}_${bytesToHex(key)}`) ?? new Uint8Array(0)
```

Comment:
Perhaps we could create a helper for `storageKeyFormatting` that parsed an address + key into the expected string? This could be helpful notably if we need to switch around the format (pre and post verkle comes to mind), then we don't have to modify it in multiple places.

Reply:
Will keep for now, we can still easily do later on